src/main/java/org/broadinstitute/consent/http/service/MatchService.java

```diff
@@ -228,15 +209,15 @@ private Match singleEntitiesMatch(Consent consent, DataAccessRequest dar) {
             logger.error("Data Access Request is null");
             throw new IllegalArgumentException("Data Access Request cannot be null");
         }
-        Match match = createMatch(consent.getConsentId(), dar.getReferenceId(), false, false);
+        Match match;
         RequestMatchingObject requestObject = createRequestObject(consent, dar);
         String json = new Gson().toJson(requestObject);
         Response res = matchServiceTarget.request(MediaType.APPLICATION_JSON).post(Entity.json(json));
         if (res.getStatus() == Response.Status.OK.getStatusCode()) {
             ResponseMatchingObject entity = res.readEntity(rmo);
-            if (entity.isResult()) {
-                match.setMatch(true);
-            }
+            match = createMatch(consent.getConsentId(), dar.getReferenceId(), false, entity.isResult());
+        } else {
+            match = createMatch(consent.getConsentId(), dar.getReferenceId(), true, false);
         }
```

Comment:
Original code was a little convoluted. It created a match with default values and then modified them if certain conditions were met. This is more functional in that we create the object with the correct values up front instead of doing conditional modifications.